lib/ui/tab-navigation.tsx

```diff
@@ -0,0 +1,144 @@
+  if (!asChild)
+    return typeof content === "function" ? content(children) : content
+
+  const firstChild = React.Children.only(children) as React.ReactElement
+  return React.cloneElement(firstChild, {
+    children:
+      typeof content === "function"
+        ? content(firstChild.props.children)
+        : content,
+  })
+}
+
+interface TabNavigationProps
+  extends Omit<
+    React.ComponentPropsWithoutRef<typeof NavigationMenuPrimitives.Root>,
+    "orientation" | "defaultValue" | "dir"
+  > {
+  type?: "primary" | "secondary"
+}
+
+const TabNavigation = React.forwardRef<
+  React.ElementRef<typeof NavigationMenuPrimitives.Root>,
+  TabNavigationProps
+>(({ className, children, type = "primary", ...props }, forwardedRef) => (
+  <NavigationMenuPrimitives.Root ref={forwardedRef} {...props} asChild={false}>
+    <NavigationMenuPrimitives.List
+      className={cn(
+        // base
+        "flex items-center justify-start gap-1 overflow-x-auto whitespace-nowrap border border-solid px-6 py-3 [scrollbar-width:none] [&::-webkit-scrollbar]:hidden",
+        // background
+        type === "primary"
+          ? "bg-f1-background-transparent"
+          : "bg-f1-background-secondary/25",
+        // border
+        "border-b border-transparent",
+        // border color
+        "border-b-f1-border-secondary",
+        className
+      )}
+    >
+      {children}
+    </NavigationMenuPrimitives.List>
+  </NavigationMenuPrimitives.Root>
+))
+
+TabNavigation.displayName = "TabNavigation"
+
+interface TabNavigationLinkProps
+  extends Omit<
+    React.ComponentPropsWithoutRef<typeof NavigationMenuPrimitives.Link>,
+    "onSelect"
+  > {
+  disabled?: boolean
+  active?: boolean
+  type?: "primary" | "secondary"
+}
+
+const TabNavigationLink = React.forwardRef<
+  React.ElementRef<typeof NavigationMenuPrimitives.Link>,
+  TabNavigationLinkProps
+>(
+  (
+    {
+      asChild,
+      disabled,
+      active,
+      className,
+      children,
+      type = "primary",
+      ...props
+    },
+    forwardedRef
+  ) => (
+    <NavigationMenuPrimitives.Item className="flex" aria-disabled={disabled}>
+      <NavigationMenuPrimitives.Link
+        data-active={active ? "true" : undefined}
+        aria-disabled={disabled}
+        className={cn(
+          "group relative flex shrink-0 select-none items-center justify-center rounded-xl no-underline focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-f1-ring focus-visible:ring-offset-1",
+          disabled ? "pointer-events-none" : ""
+        )}
+        ref={forwardedRef}
+        onSelect={() => {}}
+        asChild={asChild}
+        {...props}
+      >
+        {getSubtree({ asChild, children }, (children) => (
+          <span
+            className={cn(
+              // base
+              "flex items-center justify-center whitespace-nowrap rounded-xl px-3 py-1.5 font-medium transition-all",
+              // background
+              type === "primary"
```

Comment:
let's maybe gather all the classes for primary and secondary to an object and inject it all together?

```ts
const classes = {
  primary: "bg-f1-background-transparent group-hover:bg-f1-background-secondary group-hover:text-f1-foreground",
secondary: "bg-f1-background/60 group-hover:border-f1-border"
}
```

Then we can get it by the type variable

Reply:
Done with `cva`, not sure if it looks cleaner now though that secondary is a boolean 😅